cmd/gazelle/fix-update.go

```diff
@@ -372,12 +372,6 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 		}
 	})
 
-	for _, lang := range languages {
-		if finishable, ok := lang.(language.FinishableLanguage); ok {
-			finishable.DoneGeneratingRules()
-		}
-	}
-
 	if len(errorsFromWalk) == 1 {
 		return errorsFromWalk[0]
```

Comment:
This now changes the semantics from "always called" to "only called if the walk didn't return any errors" - I don't have a problem with the call moving later, but it _is_ important to me that it's unconditionally called.

Otherwise, I have no preference for how many hooks or what they look like :) Thanks!